### PR TITLE
binlog: use pumps client to write binlog

### PR DESCRIPTION
**Pull request by WangXiangUSTC**
### What problem does this PR solve? 
use pumps client to write binlog

### What is changed and how it works?
checkout from master, and then 
```
git cherry-pick 71d13534aabe07d42b42681531a83b5fb5b6a407
```
resolve some conflict.

this commit is reviewed before, you can see https://github.com/pingcap/tidb/pull/7659

### Check List 

Related changes
 - Need to update the documentation
 - Need to update the `tidb-ansible` repository
 - Need to be included in the release note

### CODE REVIEW

File: sessionctx/binloginfo/binloginfo.go [hunk 1]
```diff
@@ -111,58 +115,47 @@ func (info *BinlogInfo) WriteBinlog(clusterID uint64) error {
...
-			// This kind of error is not critical and not retryable, return directly.
-			return errors.Errorf("binlog data is too large (%s)", err.Error())
-		}
-		log.Errorf("write binlog error %v", err)
-		time.Sleep(time.Second)
+	if info.Client == nil {
+		return errors.New("pumps client is nil")
 	}
 
+	// will retry in PumpsClient if write binlog fail.
```
Comment (zimulala) [inline sessionctx/binloginfo/binloginfo.go]: s/will/It will

File: sessionctx/binloginfo/binloginfo.go [hunk 2]
```diff
@@ -14,53 +14,57 @@
...
 	"google.golang.org/grpc"
 )
 
 func init() {
 	grpc.EnableTracing = false
+	// don't need output pumps client's log
+	pumpcli.Logger.Out = ioutil.Discard
 }
 
 var binlogWriteTimeout = 15 * time.Second
```
Comment (zimulala) [inline sessionctx/binloginfo/binloginfo.go]: Could we remove this variable and the function of `SetGRPCTimeout`? It seems we don't use them anymore.

File: session/tidb_test.go [hunk 3]
```diff
@@ -125,21 +123,6 @@ func (s *testMainSuite) TestRetryOpenStore(c *C) {
 	c.Assert(uint64(elapse), GreaterEqual, uint64(3*time.Second))
 }
 
-func (s *testMainSuite) TestRetryDialPumpClient(c *C) {
```
Comment (GregoryIan) [inline session/tidb_test.go]: remove this testing？ if all pumps were down, what would tidb happen?

Comment (WangXiangUSTC) [reply session/tidb_test.go]: it is moved to pumps client, pumps client will retry connect to pump

Comment (GregoryIan) [reply session/tidb_test.go]: if so, it wouldn't return a response, right? is it reasonable?

Comment (WangXiangUSTC) [reply session/tidb_test.go]: this test is create grpc connection to pump directly, but this is implement in pump client now, so remove DialPumpClientWithRetry function, and don't need this unit test anymore.

Comment (GregoryIan) [reply session/tidb_test.go]: no, you should add it later. from tidb perspective,  it should know what would happen if there are no available pumps

**Comment by GregoryIan** [comment]
@tiancaiamao PTAL

File: session/tidb_test.go (hunk 3 above)
Comment (GregoryIan) [reply session/tidb_test.go]: please add a jira issue for it

Comment (WangXiangUSTC) [reply session/tidb_test.go]: OK， I will create a lira issue

Comment (GregoryIan) [reply session/tidb_test.go]: https://internal.pingcap.net/jira/browse/TOOL-518

**Pull request merged**